sdl_android/src/androidTest/java/com/smartdevicelink/transport/TransportBrokerTest.java

```diff
@@ -27,68 +28,85 @@ private void sleep(){
 	}
 	
 	public void testStart(){
-		TransportBroker broker = new TransportBroker(mContext, SdlUnitTestContants.TEST_APP_ID,rsvp.getService());
-		assertTrue(broker.start());
-		broker.stop();
+		new Handler(Looper.getMainLooper()).post(new Runnable() {
```

Comment:
I don't believe these all have to be on the main thread. I think they just need a looper. You will have to call looper.prepare. The reason it works with putting them on the main thread is that the main thread's looper is already prepared.